### PR TITLE
Add a button to the UI for exporting Osquery results to CSV

### PR DESCRIPTION
**Pull request by simstoykov**

### CODE REVIEW

**Comment by simstoykov** [comment]
Below you can see that the button is not displayed when the flow encountered an error, and it is displayed if it finished successfully with results:
image

File: grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ts [hunk 1]
```diff
@@ -80,6 +80,24 @@ export class OsqueryDetails extends Plugin {
...
+          return `/api/clients/${clientId}/flows/${flowId}/osquery-results/CSV`;
+        } else {
+          return null;
+        }
+      }),
+      filter(isNonNull),
+  );
+
+  readonly atLeastOneRowAvailable$ = this.displayTable$.pipe(
+      map(table => table.rows?.length),
```
Comment (max-vogler) [inline grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ts:98]: `atLeastOneRowAvailable` sounds like a boolean, but the underlying type seems to be number. Please rename if you'd like to keep the type, or convert to a boolean if you'd like to keep the name.

File: grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details_test.ts [hunk 2]
```diff
@@ -246,4 +249,74 @@ describe('osquery-details component', () => {
...
+              newOsqueryTable('doesnt matter', ['column'], [['cell']]),
+        }),
+      ],
+    };
+
+    const fixture = createFixtureFrom(testFlowListEntry);
+    const parsedElements = new OsqueryDetailsDOM(fixture.debugElement);
+
+    expect(parsedElements.exportCsvButton).toBeTruthy();
+    expect(parsedElements.exportCsvButtonText).toBe('Download results as CSV');
```
Comment (max-vogler) [inline grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details_test.ts:320]: Please test that the link contains the correct URL.

File: grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ng.html [hunk 3]
```diff
@@ -10,6 +10,18 @@
 </div>
 
 <ng-container *ngIf="flowCompleted$ | async">
+  <div
+    *ngIf="atLeastOneRowAvailable$ | async"
+    class="flow-details-summary-actions export-button">
+    <a
```
Comment (max-vogler) [inline grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ng.html:16]: Please add the [`download`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) attribute.

**Review by max-vogler — CHANGES_REQUESTED** [review]

**Comment by simstoykov** [comment]
@max-vogler PTAL

**Review by max-vogler — APPROVED** [review]

**Pull request merged**